addons/product/tests/test_product_attribute_value_config.py

```diff
@@ -697,3 +697,19 @@ def test_inactive_related_product_update(self):
         product.action_unarchive()
         self.assertTrue(product.active, 'The product should be unarchived.')
         self.assertEqual(product_attribut.number_related_products, 0, 'The product attribute must not have an associated product')
+
+    def test_copy_extra_prices_of_product_attribute_values(self):
+        """
+            Checks that the price_extra of the values of an attribute are copied
+            when the product is copied.
+        """
+        # Check for copied product template
+        product_template = self.computer
+        extra_prices = product_template.attribute_line_ids.product_template_value_ids.mapped('price_extra')
+        copied_template = product_template.copy()
+        copied_extra_prices = copied_template.attribute_line_ids.product_template_value_ids.mapped('price_extra')
+        self.assertEqual(extra_prices, copied_extra_prices)
+        # Check for copied variant
+        product_variant = product_template.product_variant_ids[0]
+        copied_variant = product_variant.copy()
+        self.assertEqual(product_variant.lst_price, copied_variant.lst_price)
```

Comment:
https://github.com/odoo/odoo/blob/dda9696f35afded8897770799424c3af033df357/addons/product/models/product.py#L458-L463

So I'm not sure this is really necessary =]